### PR TITLE
Support storybook 6.4

### PR DESCRIPTION
**Pull request by amitzur**
[Trello](https://trello.com/c/twTRnZtb/1159-support-storybook-64)
Storybook version 6.4 (released November 26th 2021) breaks an internal API that we used to render stories.
Instead of using `__STORYBOOK_STORY_STORE__.setSelection` we should now do:
```
__STORYBOOK_PREVIEW__.urlStore.setSelection({storyId});
await __STORYBOOK_PREVIEW__.renderSelection();
```

Note that the API is now async. Storybook loads the CSF file and runs it within this function.

Another small break is how we detect that storybook is started. The String we relied on used to be something like `Storybook 6.3.12 started`. But now it's `Storybook 6.4.0 for React started`. I don't expect other frameworks to behave much differently (e.g. Vue and Angular). And the change was very small, I'm not even sure why the regex was what it was.

Current status:
- test **enforces legacy concurrency** fails with expected `maxRunning` to be 9 instead of 10. I have no idea how this is connected to storybook 6.4, so it needs investigation.

### CODE REVIEW

**Comment by SophieDePaula** [comment]
In `storybook` version `6.4` onwards, the `body` element in the `iFrame` where the component is rendered now has `padding: 1rem` , before we had `margin: 8px`. That caused visual changes throughout our visual tests, therefore, I had to update some baselines and some tests.
I still have a few tests that I need to update, but the` enforces legacy concurrency ` test was not failing for me locally.

**Review by SophieDePaula — APPROVED** [review]
All the tests should pass now

**Comment by SophieDePaula** [comment]
Here are the tests I needed to update due to the visual changes `storybook 6.4` presented:

- [Global query parames](https://eyes.applitools.com/app/test-results/00000251762869832400/?accountId=UAujt6tHnEKUivQXIz7G6A~~)
- [Simple Storybook: Single story](https://eyes.applitools.com/app/test-results/00000251762869792207/?accountId=UAujt6tHnEKUivQXIz7G6A~~)
- [Fake IE story](https://eyes.applitools.com/app/test-results/00000251762869764970/?accountId=UAujt6tHnEKUivQXIz7G6A~~)
- [Accessibility storybook](https://eyes.applitools.com/app/test-results/00000251762869623121/?accountId=UAujt6tHnEKUivQXIz7G6A~~)
- [visualGridOptions storybook](https://eyes.applitools.com/app/test-results/00000251762869589712/?accountId=UAujt6tHnEKUivQXIz7G6A~~)
- [Wait before capture](https://eyes.applitools.com/app/test-results/00000251762869539986/?accountId=UAujt6tHnEKUivQXIz7G6A~~)
- [Layout breakpoints](https://eyes.applitools.com/app/test-results/00000251762867726230/?accountId=UAujt6tHnEKUivQXIz7G6A~~)
- [Layout breakpoints](https://eyes.applitools.com/app/test-results/00000251762867690223/?accountId=UAujt6tHnEKUivQXIz7G6A~~)
- [Cross-origin-iframe storybook](https://eyes.applitools.com/app/test-results/00000251762860996953/?accountId=UAujt6tHnEKUivQXIz7G6A~~), here I had to create a baseline with a bigger viewport as [this was the result after the upgrade](https://eyes.applitools.com/app/test-results/00000251762869914256/00000251762869913818/steps/1?accountId=UAujt6tHnEKUivQXIz7G6A~~&mode=step-editor)
- [Simple storybook](https://eyes.applitools.com/app/test-results/00000251762860805383/00000251762860805164/steps/1?accountId=UAujt6tHnEKUivQXIz7G6A~~&mode=step-editor), here I had to change the background color to be on a specific `div` instead the `html`

**Review by amitzur — COMMENTED** [review]
@SophieDePaula looks very good.
I thought at first that a better change for the cross-origin test would be to instead of changing the test to be at 810 width, just lower the font-size a bit so that the text fits in the 640 width.
But then I understood that it affect other custom coverage test (`TestCheckDomSnapshotCrossOriginFrames_VG`) so I reverted it back to 810 like you did.

See one question inline

File: packages/eyes-storybook/test/it/getStories.it.test.js [hunk 1]
```diff
@@ -53,39 +62,124 @@ describe('getStories', () => {
...
-          {name: 'story 1.1', kind: 'Nested/Component', parameters},
-          {name: 'story 1.2', kind: 'Nested/Component', parameters},
-          {name: 'a yes-a b', kind: 'Button with-space yes-indeed', parameters},
+          {
+            name: 'image',
+            kind: 'Image',
+            parameters: {
+              ...parameters,
+              __id: 'image--image',
+              fileName: './test/fixtures/appWithStorybook/index.js-2',
```
[the remark this answers is not in the document]
Comment (SophieDePaula) [reply packages/eyes-storybook/test/it/getStories.it.test.js:71]: Yes, for every `kind` this number increases

**Pull request merged**